src/infomgmag/mars/CountryAgent.java

```diff
@@ -25,8 +26,20 @@ public Territory getTerritory() {
         return territory;
     }
 
-    public void calculateOwnershipValue(Double friendliesweight, Double enemyweight, Double farmiesweight, Double earmiesweight) { //calculates value of owning a territory, TODO: the actual final calculation has more factors includings continents and such
-        this.value = (((friendlyNeighbours() * friendliesweight) + (enemyNeighbours() * enemyweight) + (friendlyArmies() * farmiesweight) + (enemyArmies() * earmiesweight)));
+    public void calculateOwnershipValue(Double friendliesweight, Double enemyweight, Double farmiesweight, Double earmiesweight
+        ,Double continentBorderWeight, Double ownWholeContinentWeight, Double enemyOwnsWholeContinentWeight, Double percentageOfContinentWeight) { //calculates value of owning a territory
+
+        Double territoryvalue = 0.0;
+        int x = ownWholeContinent() ? 1 : 0;
```

Comment:
The entire idea of these things is that you don't have to declare an additional variable, right?

Reply:
Done